rust/candid/src/parser/value.rs

```diff
@@ -334,6 +334,8 @@ pub mod pretty {
         enclose_space("{", fs, "}")
     }
 
+    const MAX_VEC_ELEMENTS: usize = 20;
```

Comment:
This means that you cannot really use the pretty-printed format in scripts any more, as you lose data. Is that a good idea? Does the library give applications an option whether to shorten values this way?

Also, when shortening vec, why ont `blob` similarly (we have messages with pretty large blobs…)

Reply:
It is shortening both `vec` and `blob`. Pretty printing is mostly for human consumption. Large vector also makes the layout algorithm slow for pretty printer. I should have a non-pretty printed version for scripting.

Reply:
Done in https://github.com/dfinity/candid/pull/118.